core-bundle/src/DataContainer/DataContainerOperation.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\DataContainer;
+
+use Contao\DataContainer;
+use Contao\StringUtil;
+
+class DataContainerOperation implements \ArrayAccess
```

Comment:
```suggestion
/**
 * @implements \ArrayAccess<string, string>
*/
class DataContainerOperation implements \ArrayAccess
```

Reply:
I'm not sure about `mixed` - took that from your `offsetGet()` return type. If it should actually be `string`, we should change it at both places and maybe throw an error in `offsetSet()` if something else was set.

IMHO a `/** param array<…> $operation /*` type should also be added to the constructor.